### PR TITLE
Assorted portability fixes

### PR DESCRIPTION
**Pull request by algernon**
This includes a small batch of fixes aimed at making it easier to port Kaleidoscope to different Arduino Cores, perhaps even ones that do not support KeyboardioHID.

Included changes are:
- We should not redefine `new` on STM32 (used by LEDModeManager). There might be better ways to do the same thing, this was easiest.
- `driver::hid::Keyboardio` has been restricted to not compile when using a core that does not support PluggableUSB.
- `device::Base` has been switched to use `hid::Base` as default, rather than `hid::Keyboardio`, so that we can support different HID drivers. All appropriate place have been updated to explicitly set `hid::Keyboardio`.
- Moved `driver::hid::mouse::stop` from the base class to the specific implementation, to avoid a dependency on KeyboardioHID in the base class.
- `driver::hid::mouse::getReport()` has been removed for the same reason. There are no past or present users, and it shouldn't have been exposed in the first place.
- The base HID driver defines `HID_BOOT_PROTOCOL` if it is undefined, to remove a dependency on KeyboardioHID.

Basically, these changes make it not only theoretically, but practically possible to use a different HID driver, if need be. The LEDModeManager is unrelated to HID, but still a portability bug, so I lumped it in here.

### CODE REVIEW

File: src/kaleidoscope/driver/hid/Keyboardio.h [hunk 1]
```diff
@@ -16,6 +16,9 @@
  */
 
 #pragma once
+
+#ifndef USBD_USE_HID_COMPOSITE
```
Comment (gedankenexperimenter) [inline src/kaleidoscope/driver/hid/Keyboardio.h:20]: Please forgive me if I'm being naïve, but this strikes me as something that ought to be handled by not including this header file if `USBD_USE_HID_COMPOSITE` is defined, rather than excluding it here.  I have a lot of difficulty navigating the maze of headers used by the hardware modules, so I suppose you've got a good reason for doing it this way, but I think a comment explaining why this is here in the code is warranted.

**Review by gedankenexperimenter — COMMENTED** [review]
It looks like maybe 01a8d2a6 is obviated by 5465316e, but I don't know (nor do I know how to test it, at least off the top of my head).  Do we have any existing sketches for a board that doesn't use KeyboardioHID?

**Comment by algernon** [comment]
> It looks like maybe [01a8d2a](https://github.com/keyboardio/Kaleidoscope/commit/01a8d2a6e50ea03a8f2b9aeb0e4602d51ca0ebbf) is obviated by [5465316](https://github.com/keyboardio/Kaleidoscope/commit/5465316e170d832492b76894cc7425da15b7c0ef), but I don't know (nor do I know how to test it, at least off the top of my head). Do we have any existing sketches for a board that doesn't use KeyboardioHID?

There's the `wip/planck-ez` branch where you can test, albeit it's a tad cumbersome to do so. But you _are_ correct, the latter commit makes the former irrelevant indeed. The reason I added the ifdef guard is that Arduino insists on compiling everything anyway, even when not included. However, there's a catch that I forgot! It only does that if there's an accompanying `.cpp` file, which is not the case here. As such, Arduino ignores the header completely unless included, which we do not do anymore with the base device switching away from `hid::keyboardio`.

I'll knock that commit out. Thanks for noticing!

File: src/kaleidoscope_internal/LEDModeManager.h [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 #include <stddef.h>
 
-#ifdef KALEIDOSCOPE_VIRTUAL_BUILD
+#if defined(KALEIDOSCOPE_VIRTUAL_BUILD) || defined(ARDUINO_ARCH_STM32)
```
Comment (obra) [inline src/kaleidoscope_internal/LEDModeManager.h:26]: This *feels* like it's going to end up being a hassle longer term. But that's not a blocker to merge.

**Review by obra — APPROVED** [review]
